hoomd/hpmc/pytest/test_compute_sdf.py

```diff
@@ -0,0 +1,179 @@
+from __future__ import print_function
+from __future__ import division
+import hoomd
+import pytest
+import numpy as np
+import hoomd.hpmc.pytest.conftest
+
+
+def test_before_attaching():
+    sdf = hoomd.hpmc.compute.SDF(xmax=0.02, dx=1e-4)
+
+    assert sdf.xmax == 0.02
+    assert sdf.dx == 1e-4
+    with pytest.raises(hoomd.error.DataAccessError):
+        sdf.sdf
+    with pytest.raises(hoomd.error.DataAccessError):
+        sdf.betaP
+
+
+def test_after_attaching(valid_args, simulation_factory,
+                         lattice_snapshot_factory):
+    snap = lattice_snapshot_factory(particle_types=['A'])
```

Comment:
This test fails on the GPU for two reasons. 1) The lattice snapshot factory makes 3D simulations by default and 2) Some of the particles in `valid_args` have rather large sizes.

@danevans09 can you please fix by adding the dimensionality of the system to `valid_args` and pass it to `lattice_snapshot_factory`. Also update the other HPMC tests that use `lattice_snapshot_factory` and `two_particle_snapshot_factory` to use this parameters. Please also adjust the vertices of the shapes in `valid_args` so that the diameter of the test shapes is close to 1. One of the convex polygon shapes currently has a diameter of 9.65539.